ibm/service/contextbasedrestrictions/resource_ibm_cbr_rule.go

```diff
@@ -362,39 +308,21 @@ func resourceIBMCbrRuleCreate(context context.Context, d *schema.ResourceData, m
 
 	d.SetId(*rule.ID)
 
-	// handle Eventual consistency case
-	readNewCbrRule(contextBasedRestrictionsClient, context, d)
-
-	return resourceIBMCbrRuleRead(context, d, meta)
-}
-
-func resourceIBMCbrRuleRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	contextBasedRestrictionsClient, err := meta.(conns.ClientSession).ContextBasedRestrictionsV1()
-	if err != nil {
-		return diag.FromErr(err)
+	if fromErr := resourceIBMCbrRuleSetData(response, rule, d); fromErr != nil {
+		return diag.FromErr(fmt.Errorf("Error setting rule's resource data: %s", fromErr))
 	}
 
-	getRuleOptions := &contextbasedrestrictionsv1.GetRuleOptions{}
-
-	getRuleOptions.SetRuleID(d.Id())
-
-	rule, response, err := contextBasedRestrictionsClient.GetRuleWithContext(context, getRuleOptions)
-	if err != nil {
-		if response != nil && response.StatusCode == 404 {
-			d.SetId("")
-			return nil
-		}
-		log.Printf("[DEBUG] GetRuleWithContext failed %s\n%s", err, response)
-		return diag.FromErr(fmt.Errorf("GetRuleWithContext failed %s\n%s", err, response))
-	}
+	return nil
+}
 
-	if err = d.Set("x_correlation_id", getRuleOptions.XCorrelationID); err != nil {
+func resourceIBMCbrRuleSetData(response *core.DetailedResponse, rule *contextbasedrestrictionsv1.Rule, d *schema.ResourceData) diag.Diagnostics {
+	if err := d.Set("x_correlation_id", response.Headers.Get("x_correlation_id")); err != nil {
```

Comment:
Also for this one, isn't the header `X-Correlation-ID`.

Reply:
put `d.Set("x_correlation_id"` back to `resourceIBMCbrRuleRead` function